Sources/ScipioKit/URL+Extensions.swift

```diff
@@ -1,8 +1,28 @@
 import Foundation
-import TSCBasic
 
+/// Extensions for compatibility with TSC's Path implementation.
 extension URL {
-    var absolutePath: AbsolutePath {
-        return try! AbsolutePath(validating: path)
+    // ref: https://github.com/swiftlang/swift-tools-support-core/blob/f9b401016b70c6b8409e5c97e74d97513d1a8d02/Sources/TSCBasic/Path.swift#L598-L614
+    var dirname: String {
+        let path = self.standardizedFileURL.path(percentEncoded: false)
+        guard let idx = path.lastIndex(of: "/") else {
+            // No path separators, so directory is current directory
+            return "."
+        }
+        // If the only separator is the first character, it's the root directory
+        if idx == path.startIndex {
+            return "/"
+        }
+        // Otherwise, return everything up to (but not including) the last separator
+        return String(path.prefix(upTo: idx))
+    }
+
+    // ref: https://github.com/swiftlang/swift-tools-support-core/blob/f9b401016b70c6b8409e5c97e74d97513d1a8d02/Sources/TSCBasic/Path.swift#L661-L663
+    var parentDirectory: URL {
+        pathComponents.isEmpty ? self : deletingLastPathComponent()
```

Comment:
`URL(filePath: "/").pathComponents` returns `["/"]` so this should not be correct.